core/common/test/samples/LocalDateRangeSamples.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test.samples
+
+import kotlinx.datetime.*
+import kotlin.random.Random
+import kotlin.test.Test
+
+class LocalDateRangeSamples {
+    val Jan_01_2000 = LocalDate(2000, 1, 1)
+    val Jan_05_2000 = LocalDate(2000, 1, 5)
+    val Jan_06_2000 = LocalDate(2000, 1, 6)
+
+    @Test
+    fun simpleRangeCreation() {
+        // Creating LocalDateRange from LocalDate
+        check(Jan_01_2000..Jan_05_2000 == LocalDateRange(Jan_01_2000, Jan_05_2000))
+        check(Jan_01_2000..<Jan_06_2000 == LocalDateRange(Jan_01_2000, Jan_05_2000))
+        check(Jan_05_2000 downTo Jan_01_2000 == LocalDateProgression(Jan_05_2000, Jan_01_2000, -1))
```

Comment:
Samples shouldn't contain internal API (like `LocalDateProgression` constructor here). Suggestion: how about something like `check((LocalDate(1, 1, 2000)..LocalDate(1, 5, 2000)).toList() == listOf(...))` instead? It's verbose but explains in no uncertain terms what ranges contain, which is what we want from samples.